flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/delegation/PlannerBase.scala

```diff
@@ -175,12 +175,10 @@ abstract class PlannerBase(
       case catalogSink: CatalogSinkModifyOperation =>
         val input = getRelBuilder.queryOperation(modifyOperation.getChild).build()
         val identifier = catalogManager.qualifyIdentifier(catalogSink.getTablePath: _*)
-        getTableSink(identifier).map(sink => {
-          TableSinkUtils.validateSink(catalogSink, identifier, sink)
+        getTableSink(identifier).map { case (table, sink) =>
+          TableSinkUtils.validateSink(catalogSink, identifier, sink, table.getPartitionKeys)
           sink match {
-            case partitionableSink: PartitionableTableSink
-              if partitionableSink.getPartitionFieldNames != null
-                && partitionableSink.getPartitionFieldNames.nonEmpty =>
+            case partitionableSink: PartitionableTableSink =>
               partitionableSink.setStaticPartition(catalogSink.getStaticPartitions)
```

Comment:
Maybe just for the test in `PartitionableSinkITCase.testInsertWithStaticPartitions`, it get `staticPartitions` from the origin `Sink`, if in `ExecSinkRule`, it will be another sink.
I think we can modify this in https://github.com/apache/flink/pull/9796